samples/sample_common/src/sysmem_allocator.cpp

```diff
@@ -392,11 +392,20 @@ mfxStatus SysMemBufferAllocator::AllocBuffer(mfxU32 nbytes, mfxU16 type, mfxMemI
         return MFX_ERR_UNSUPPORTED;
 
     mfxU32 header_size = MSDK_ALIGN32(sizeof(sBuffer));
-    mfxU8 *buffer_ptr = (mfxU8 *)calloc(header_size + nbytes + 32, 1);
+
+//MSVS still has no aligned_alloc support, because
+//it incompatible with their implementation of free()
+#if defined (LINUX32) || defined (LINUX64)
+    mfxU8 *buffer_ptr = (mfxU8 *)aligned_alloc(sysconf(_SC_PAGESIZE), header_size + nbytes + 32);
+#else
+    mfxU8 *buffer_ptr = (mfxU8 *)malloc(header_size + nbytes + 32);
```

Comment:
hmmm. What is the header_size? Actually, the critical point is: where pointer to the surface data starts? This pointer should be aligned by page boundary. If it is not (but header is) - this will NOT do.

Reply:
Good catch! @AntonGrishin , can you please check if we need this header at all? Better to  get rid of it.